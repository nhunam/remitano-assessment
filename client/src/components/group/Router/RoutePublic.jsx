import React from 'react';
import PropTypes from 'prop-types';
import { Route, Redirect } from 'react-router-dom';
import { AppConfig } from 'configurations';

export const RoutePublic = ({ component: Component, isAuthenticated, to, ...rest }) => (
  <Route
    {...rest}
    render={props => (isAuthenticated ? <Redirect to={to} /> : <Component {...props} />)}
  />
);

RoutePublic.propTypes = {
  component: PropTypes.oneOfType([PropTypes.func, PropTypes.object]).isRequired,
  isAuthenticated: PropTypes.bool.isRequired,
  to: PropTypes.string,
};

RoutePublic.defaultProps = {
  to: AppConfig.routePrefix,
};

export default RoutePublic;
