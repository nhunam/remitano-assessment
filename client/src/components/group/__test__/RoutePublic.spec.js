import { MemoryRouter } from "react-router-dom";
import NotFoundScreen from "containers/screens/NotFound";
import React from "react";
import RoutePublic from "components/group/Router/RoutePublic";
import { renderToString } from "react-dom/server";

describe("Route Public", () => {
  it("should redirect for unauthenticated access", () => {
    const render = renderToString(
      <MemoryRouter initialEntries={["/private"]}>
        <RoutePublic isAuthenticated={false} component={NotFoundScreen} />
      </MemoryRouter>
    );

    expect(render).toMatchSnapshot();
  });

  it("should redirect for authenticated access", () => {
    const render = renderToString(
      <MemoryRouter initialEntries={["/private"]}>
        <RoutePublic isAuthenticated={true} component={NotFoundScreen} />
      </MemoryRouter>
    );

    expect(render).toMatchSnapshot();
  });
});
