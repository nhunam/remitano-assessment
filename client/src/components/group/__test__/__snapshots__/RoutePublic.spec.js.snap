// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Route Public should redirect for authenticated access 1`] = `""`;

exports[`Route Public should redirect for unauthenticated access 1`] = `"<div class=\\"sc-bdVaJa bDWFJH 404-error-page\\"><div class=\\"error-content\\"><h1>404</h1><h3>Looks like you got lost</h3><p>The page youre looking for doesnt exist or has been moved</p></div></div>"`;
